lib/robots/dor_repo/gis_assembly/finish_gis_assembly_workflow.rb

```diff
@@ -29,21 +24,22 @@ def perform(druid)
             metadata/geoMetadata.xml
           ].each do |f|
             fn = File.join(rootdir, f)
-            raise "finish-gis-assembly-workflow: #{druid} is missing required file: #{fn}" unless File.size?(fn)
+            raise "finish-gis-assembly-workflow: #{bare_druid} is missing required file: #{fn}" unless File.size?(fn)
           end
 
           # delete all staged files in temp/
           tmpdir = "#{rootdir}/temp"
           if File.directory?(tmpdir)
-            LyberCore::Log.debug "finish-gis-assembly-workflow deleting #{tmpdir}"
+            logger.debug "finish-gis-assembly-workflow deleting #{tmpdir}"
             FileUtils.rm_r(tmpdir)
           end
 
           # load workspace with identical copy of stage
-          destdir = GisRobotSuite.locate_druid_path druid, type: :workspace
+          destdir = GisRobotSuite.locate_druid_path bare_druid, type: :workspace
           FileUtils.mkdir_p(destdir) unless File.directory?(destdir)
-          LyberCore::Log.info "finish-gis-assembly-workflow: #{druid} migrating object to #{destdir}"
-          system("rsync -av --delete #{rootdir}/ #{destdir}/")
+          logger.info "finish-gis-assembly-workflow: #{bare_druid} migrating object to #{destdir} from #{rootdir}"
+          FileUtils.cp_r("#{rootdir}/.", destdir)
+          FileUtils.rm_f(rootdir)
```

Comment:
my recollection from talking with andrew is that while `rsync` can use checksums as a method for determining whether a target file needs to be updated, it doesn't checksum to confirm that the target and source match after completing the copy.

one other question that occurred to me: how much do symlinks and file perms matter?  i'm guessing perms won't be an issue, since integration testing and regular usage would probably suss that out, as happened when testing the pres robots switch from tarpipe (which broke goobi and web archive accessioning at first).  no idea whether integration tests exercise symlink usage.  but fwiw, the `-a` option for `rsync` says `archive mode; same as -rlptgoD (no -H)`.  `-l` is `copy symlinks as symlinks`, `-p` preserves perms, `-t` preserves timestamps, `-g` preserves group, `-o` owner, and `-D` device files (i'd be surprised if `t` or `d` matters, and would be surprised if `o` and `g` mattered but went uncaught by the integration tests and regular usage).

i would bet this change is not a problem in practice, but am not totally sure.

Reply:
The rsync wasn't working, which is why I replaced it.

Integration tests do pass. If we encounter a problem that was not caught, then this can be re-visited.

Someone made reference to an earlier developer who used rsync for everything, so it might just be that.

Reply:
> The rsync wasn't working, which is why I replaced it.

that seems like a good reason to replace 🙂 

> Integration tests do pass. If we encounter a problem that was not caught, then this can be re-visited.

👍 

> Someone made reference to an earlier developer who used rsync for everything, so it might just be that.

i also wondered if that might be the explanation